File: lib/js_test_server/server/resources/framework_file.rb
class JsTestServer::Server::Resources::FrameworkFile < JsTestServer::Server::Resources::File
  map "/framework"

  get "/?" do
    do_get
  end

  get "*" do
    do_get
  end

  def absolute_path
    @absolute_path ||= ::File.expand_path("#{framework_path}#{relative_path.gsub(%r{^/framework}, "")}")
  end
end

File: lib/js_test_server/server/resources/framework_file.js
var sys = require('sys'),
    express = require('express'),
    path = require('path');

exports.bind = function() {
  get("/framework/*", function() {
    JsTestServer.Server.Resources.FileDispatch(
      this,
      path.join(
        JsTestServer.Server.frameworkPath,
        this.url.pathname.replace(/\/framework\//, "/")
      )
    )
  });
}
